src/Microsoft.ML.Data/Transforms/CopyColumnsTransform.cs

```diff
@@ -3,25 +3,117 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
-using System.Reflection;
+using System.Collections.Generic;
+using System.Linq;
 using System.Text;
+using Microsoft.ML.Core.Data;
 using Microsoft.ML.Runtime;
 using Microsoft.ML.Runtime.CommandLine;
 using Microsoft.ML.Runtime.Data;
 using Microsoft.ML.Runtime.EntryPoints;
 using Microsoft.ML.Runtime.Internal.Utilities;
 using Microsoft.ML.Runtime.Model;
 
-[assembly: LoadableClass(CopyColumnsTransform.Summary, typeof(CopyColumnsTransform), typeof(CopyColumnsTransform.Arguments), typeof(SignatureDataTransform),
-    CopyColumnsTransform.UserName, "CopyColumns", "CopyColumnsTransform", CopyColumnsTransform.ShortName, DocName = "transform/CopyColumnsTransform.md")]
+[assembly: LoadableClass(CopyColumnsTransformer.Summary, typeof(IDataTransform), typeof(CopyColumnsTransformer),
+    typeof(CopyColumnsTransformer.Arguments), typeof(SignatureDataTransform),
+    CopyColumnsTransformer.UserName, "CopyColumns", "CopyColumnsTransform", CopyColumnsTransformer.ShortName,
+    DocName = "transform/CopyColumnsTransformer.md")]
 
-[assembly: LoadableClass(CopyColumnsTransform.Summary, typeof(CopyColumnsTransform), null, typeof(SignatureLoadDataTransform),
-    CopyColumnsTransform.UserName, CopyColumnsTransform.LoaderSignature)]
+[assembly: LoadableClass(CopyColumnsTransformer.Summary, typeof(IDataView), typeof(CopyColumnsTransformer), null, typeof(SignatureLoadDataTransform),
```

Comment:
>SignatureLoadDataTransform [](start = 121, length = 26)

Actually try removing this attribute: I feel that it's not necessary, and instead the one at line 28 is needed

Reply:
what about old models? they still contain pipelines with CopyColumn which is DataTransform.

---
In reply to: [212469990](https://github.com/dotnet/machinelearning/pull/706#discussion_r212469990) [](ancestors = 212469990)

Reply:
Right... I guess we'll have to keep it then. :(

---
In reply to: [212473044](https://github.com/dotnet/machinelearning/pull/706#discussion_r212473044) [](ancestors = 212473044,212469990)